js/counter.js

```diff
@@ -1,16 +1,97 @@
 var PLAYER = {
-  points: 0
+  points: 0,
+  teenagers: 0,
+  kettles: 0,
+  theaters: 0
 }
 
 function earnPoints() {
   PLAYER.points += 1
   updateUI()
 }
 
+function buyTeenagers(num) {
+  teenagerIncrease = () => {
+    PLAYER.teenagers += num
+    updateTeenagerDorm()
+  }
+  return teenagerIncrease
+}
+
+function buyKettles(num) {
+  kettleIncrease = () => {
+    PLAYER.kettles += num
+    updateKettleStorage()
+  }
+  return kettleIncrease
+}
+
+function buyTheaters(num) {
+  theaterIncrease = () => {
+    PLAYER.theaters += num
+    updateTheaterLocations()
+  }
+  return theaterIncrease
+}
+
+
 function updateUI() {
   $('#points').text(PLAYER.points)
 }
 
+function updateTeenagerDorm() {
+ $('#teenagers').text(PLAYER.teenagers)
+}
+
+function updateKettleStorage() {
+$('#kettles').text(PLAYER.kettles)
+}
+
+function updateTheaterLocations() {
+ $('#theaters').text(PLAYER.theaters)
+}
```

Comment:
Same as above - each of these functions has _very similar_ behavior. We could refactor this code into a more powerful `updateUI` function that takes the value/id as a parameter, e.g.:

```js
function updateUI(component) {
  const selector = '#' + component
  $(selector).text(PLAYER[component])
}
```